modules/src/crypto/mod.rs

```diff
@@ -8,10 +10,40 @@ pub(crate) fn keccak256(input: &str) -> [u8; 32] {
     result.try_into().unwrap()
 }
 
+/// This function is used to recover the public key from the signature.
+/// The signature needs to be in libsecp256k1 format.
+/// hash is the hash of the message that was signed.
+/// recovery_id is the recovery id of the signature (0 to 3).
+#[allow(dead_code)]
+fn ecrecover(hash: &[u8; 32], signature: &[u8; 64], recovery_id: u8) -> Result<PublicKey, Error> {
```

Comment:
Based on what we discussed earlier, you can remove this in favour of signature verification.